### DISCUSSION TITLE
Suggestion to explain why arrays cannot be dynamic in the "Built-in Types" guide of the Sway Book

### DISCUSSION THREAD

**Discussion by PaulRBerg** [FuelLabs/sway#3667] [Ideas]
The [Built-in Types](https://fuellabs.github.io/sway/v0.32.2/book/basics/built_in_types.html) guide in the Sway Book is great, but the [Arrays](https://fuellabs.github.io/sway/v0.32.2/book/basics/built_in_types.html#arrays) section lacks an explanation for why the arrays cannot be dynamic.

As someone coming to Fuel from Ethereum, I think that this is a severe limitation of Fuel if it doesn't offer out-of-the-box support for dynamic arrays.

**Comment by otrho** [comment]
Sway provides dynamic arrays via `std::vec::Vec` which aren't introduced until section 7.1.  So yeah, maybe a note pointing this out could be added to the Array section.

**Reply by PaulRBerg** [reply]
I agree, it would be worth cross-referencing the dynamic array `std::vec::Vec` in this context. Otherwise, everybody coming from the EVM would jump to the conclusion that dynamic arrays are not supported.

**Reply by sherloook2** [reply]
i agree with this suggestion. It would be valuable to include a reference to the std::vec::Vec dynamic array data structure in this context. Otherwise, developers transitioning from the EVM might mistakenly assume that dynamic arrays are not supported. Providing this clarification would benefit newcomers to the language.

**Reply by PaulRBerg** [reply]
@sherloook2 would you like to say something I haven't said instead of paraphrasing with ChatGPT?

Your post (and account history) smell like airdrop farming to me.

**Comment by mohammadfawaz** [comment]
Created an issue from this discussion here: https://github.com/FuelLabs/sway/issues/3841

**Comment by crystalb159** [comment]
loved it

**Comment by cheikhtahar** [comment]
i like it
